code/hsec-core/src/Security/Advisories/Core/Advisory.hs

```diff
@@ -52,9 +59,22 @@ data Advisory = Advisory
   }
   deriving stock (Show)
 
-data ComponentIdentifier = Hackage Text | GHC GHCComponent
+data ComponentIdentifier
+  = Repository RepositoryURL RepositoryName PackageName
+  | GHC GHCComponent
   deriving stock (Show, Eq)
 
+hackage :: PackageName -> ComponentIdentifier
+hackage = Repository (RepositoryURL "https://hackage.haskell.org") (RepositoryName "hackage")
+
+newtype RepositoryURL
+  = RepositoryURL { unRepositoryURL :: Text }
```

Comment:
Are you sure this shouldn't be a proper URI or URL type?

Reply:
At this point it's not that important, that being said, if you have any good library to integrate, I'd be grateful